front-end/src/app/reports/f1m/main-form/main-form.component.ts

```diff
@@ -1,10 +1,213 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, AbstractControl, Validators, FormGroup } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { takeUntil, of, from, Observable, concatAll, reduce } from 'rxjs';
+import { ValidateUtils } from 'app/shared/utils/validate.utils';
+import { schema as f1mSchema } from 'fecfile-validate/fecfile_validate_js/dist/F1M';
+import { MessageService, ConfirmationService, SelectItem } from 'primeng/api';
+import { Form1M } from 'app/shared/models/form-1m.model';
+import { TransactionTemplateMapType } from 'app/shared/models/transaction-type.model';
+import { Form1MService } from 'app/shared/services/form-1m.service';
+import { Report } from 'app/shared/models/report.model';
+import { MainFormBaseComponent } from 'app/reports/shared/main-form-base.component';
+import { PrimeOptions, LabelUtils } from 'app/shared/utils/label.utils';
+import { ContactTypeLabels, ContactTypes, Contact } from 'app/shared/models/contact.model';
+import { selectActiveReport } from 'app/store/active-report.selectors';
+import { singleClickEnableAction } from 'app/store/single-click.actions';
+import { TransactionContactUtils } from 'app/shared/components/transaction-type-base/transaction-contact.utils';
+
+const contactConfigs: { [contactKey: string]: { [formField: string]: string } } = {
+  contact_affiliated: {
+    committee_name: 'name',
+    committee_fec_id: 'committee_id',
+  },
+};
+
+const templateMapConfigs: { [contactKey: string]: { [formField: string]: string } } = {
+  contact_affiliated: {
+    committee_name: 'affiliated_committee_name',
+    committee_fec_id: 'affiliated_committee_fec_id',
+  },
+};
 
 @Component({
   selector: 'app-main-form',
   templateUrl: './main-form.component.html',
-  styleUrls: ['./main-form.component.scss']
 })
-export class MainFormComponent {
+export class MainFormComponent extends MainFormBaseComponent implements OnInit {
+  formProperties: string[] = [
+    'committee_type',
+    'filer_committee_id_number',
+    'committee_name',
+    'street_1',
+    'street_2',
+    'city',
+    'state',
+    'zip',
+    'affiliated_date_form_f1_filed',
+    'affiliated_committee_fec_id',
+    'affiliated_committee_name',
+    'statusBy',
+    'contactAffiliatedLookup',
+  ];
+  schema = f1mSchema;
+  webprintURL = '/reports/f1m/web-print/';
+  templateMap = {
+    street_1: 'street_1',
+    street_2: 'street_2',
+    city: 'city',
+    state: 'state',
+    zip: 'zip',
+  } as TransactionTemplateMapType;
+
+  committeeTypeControl: AbstractControl | null = null;
```

Comment:
use undefined?

Reply:
I went with undefined at first but was getting this typescript error because the angular form api for getting the control returns a null:

`TS2322: Type 'AbstractControl<any, any> | null' is not assignable to type 'AbstractControl<any, any> | undefined'.`

The line of code that complains is: https://github.com/fecgov/fecfile-web-app/blob/feature/1292/front-end/src/app/reports/f1m/main-form/main-form.component.ts#L105